### ISSUE TITLE
Unable to mark test messages which contain named sender information as read (aka unable to mark messages from google agenda/calendar as read)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#149]
```
What steps will reproduce the problem?
1. Receive a message for an operator-named contact. (For instance, if you go to 
http://calendar.google.com > Settings > Mobile Setup and add a phone, so you 
get a verification code.)
2. The name of the contact is set as reply contact.
3. Write "mar"

What is the expected output?
Message is marked as read.

What do you see instead?
No match for "<name>"

The contact doesn't exist, as it's a name attached to the message itself, 
rather than an actual contact name.

Additional informations:
GTalkSMS version = 2.1.3
Android OS version = 2.2
Phone model = HTC Hero

```

Original issue reported on code.google.com by `Eckankar` on 20 Apr 2011 at 11:01

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 20 Apr 2011 at 1:44
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Issue 178 has been merged into this issue.
```

Original comment by `fschm...@gmail.com` on 23 May 2011 at 11:28

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 23 May 2011 at 11:31
- Changed title: **Unable to mark test messages which contain named sender information as read (aka unable to mark messages from google agenda/calendar as read)**

**Comment by GoogleCodeExporter** [comment]
```
My issue might be in relation with this one, when using "mar" after a message 
from Google (No phone Number)  
Gtalksms says that all message from another contact are read 



smsGT
  SMS de Google :  Rappel : BioPic - Wed 22 Jun 10:30 - 11:30 (******)
  Le destinataire par défaut est Google

moi
  mar

smsGT
  Les messages de Raphaël ****** ont été marqués comme lus

```

Original comment by `lpenou.tstXX@mpo.li` on 22 Jun 2011 at 8:31

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 11 Jul 2011 at 4:08
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
I currently suspect the problem within the 
http://developer.android.com/reference/android/telephony/SmsMessage.html#getOrig
inatingAddress() method in 
https://code.google.com/p/gtalksms/source/browse/src/com/googlecode/gtalksms/rec
eivers/SmsReceiver.java#61 which may return a named string. Please check your 
logs with enabled debug log when receiving such a message for lines like:
08-16 13:31:28.367 I/gtalksms(31096): 
com.googlecode.gtalksms.action.SMS_RECEIVED: number=+491xxxxxx message=test 
roomExists=false
an check the "number" parameter if it may contains a name.
```

Original comment by `fschm...@gmail.com` on 16 Aug 2011 at 11:47

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ca777c0579a6.
```

Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 8:36
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version 3.2
```

Original comment by `Florent....@gmail.com` on 28 Aug 2011 at 3:46
- Changed state: **Deployed**
